### PR TITLE
Fixes Main Menu Visibility for Admin Mode ( #2958 )

### LINKED ISSUE #2958: Accessing General settings form Admin settings does not display disabled submenus
#### Software and hardware versions 
Collect v1.20.0, Android v9 build number 00WW_5_140, device used Nokia 8 TA-1004

#### Problem description
Accessing General settings from Admin settings does not display all General settings submenus if access to them has been revoked from normal users.

When normal user access for all settings has been disabled, only the following submenu items are available in admin mode:
1. Server Settings -> ODK Aggregate settings
2. Form management -> Form filling -> Show guidance for questions

#### Steps to reproduce the problem

1. In Admin Settings -> User Settings, uncheck all check boxes
2. Check available settings in Admin Settings -> General Settings -> Server
3. Check available settings in Admin Settings -> General Settings -> User interface
4. Check available settings in Admin Settings -> General Settings -> Form management
5. Check available settings in Admin Settings -> General Settings -> User and device identity

#### Expected behavior
Accessing General settings via the Admin settings should allow access to all the settings in General settings submenus.

#### Other information 
Class `org.odk.collect.android.preferences.DisabledPreferencesRemover` does not seem to take into account whether the user is in admin mode or not. 

Class `org.odk.collect.android.preferences.GeneralPreferencesFragment` handles admin mode correctly for main General settings categories.

### PR DESCRIPTION
**Pull request by ajwad-shaikh**
Closes #2958 

The preferences menu fragments were instantiated directly and the arguments of the previous fragment were lost. A better way to instantiate the fragments are mentioned in [this article](https://gunhansancar.com/best-practice-to-instantiate-fragments-with-arguments-in-android/)

I've implemented the best practice instantiation and the issue seems to be fixed.

#### What has been done to verify that this works as intended?
Tested on device Moto G5 Plus

#### Why is this the best possible solution? Were any other approaches considered?
The implemented method is the [best practice](https://gunhansancar.com/best-practice-to-instantiate-fragments-with-arguments-in-android/) to instantiate fragments.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
None noticed so far. Should not affect in any other way.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java [hunk 1]
```diff
@@ -64,31 +63,36 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
 
     @Override
     public boolean onPreferenceClick(Preference preference) {
-        Fragment fragment = null;
         switch (preference.getKey()) {
             case "protocol":
-                fragment = new ServerPreferences();
+                getActivity().getFragmentManager()
```
Comment (huangyz0918) [inline collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java]: Can those similar code be extracted into a method? And put an instance as parameter.

Comment (ajwad-shaikh) [reply collect_app/src/main/java/org/odk/collect/android/preferences/GeneralPreferencesFragment.java]: I have fixed the code redundancy to match the way it was done in the original code

**Comment by ajwad-shaikh** [comment]
The latest commits conform to the existing code structure and fix the code redundancy in the initial commits

**Review by grzesiek2010 — APPROVED** [review]

**Comment by jsoppela** [comment]
I've cherry-picked the commits in this PR to my fork and the issue originally reported is fixed by them. Thank you @ajwad-shaikh !

**Comment by ajwad-shaikh** [comment]
@jsoppela Always a pleasure! :smile:

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android: 4.2, 4.4, 5.1, 6.0, 7.0, 8.1
Verified cases:
- hiding all options (General Settings list is empty but Admin Settings -> General Settings contains all options)
- hiding a part of options

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"

**Comment by grzesiek2010** [comment]
Thanks @ajwad-shaikh  and @jsoppela

**Pull request merged**